controllers/pubmed.controller.ts

```diff
@@ -0,0 +1,212 @@
+import { NextApiRequest } from 'next'
+import dayjs from 'dayjs'
+import {getPublications} from './featuregenerator.controller';
+
+export async function searchPubmed(req: NextApiRequest)  {
+    console.log('req body ******************************* ', JSON.stringify(req.body));
+   return fetch(`${reciterConfig.reciterPubmed.searchPubmedCountEndpoint}`, {
+        method: "POST",
+        headers: {
+            'Content-Type': 'application/json',
+            'User-Agent': 'reciter-pub-manager-server'
+        },
+        body: JSON.stringify(req.body)
+    })
+        .then(async(res)=> {
+            if(res.status !== 200) {
+                let responseText = await res.json()
+                return {
+                    statusCode: res.status,
+                    statusText: responseText
+                }
+            } else {
+                let data: any = await res.json()
+                  if(parseInt(data, 10) <= 1000) { 
+                    return fetch(`${reciterConfig.reciterPubmed.searchPubmedEndpoint}`, {
+                        method: "POST",
+                        headers: {
+                            'Content-Type': 'application/json',
+                            'User-Agent': 'reciter-pub-manager-server'
+                        },
+                        body: JSON.stringify(req.body)
+                    })
+                    .then(async (res) => {
+                        let data: any = await res.json()
+                        let featureGeneratorData = await getPublications(req.body.personIdentifier,req);
+                        console.log('data**************************',featureGeneratorData); 
+                        console.log('Accepted/Rejected publicatioms**************************',featureGeneratorData);   
+                         let first100PubData = retrieveFirstNew100PubMedArticles(data,featureGeneratorData)
+                        return {
+                            statusCode: res.status,
+                            statusText: formatPubmedSearch(first100PubData, (data && data.length > 100?true : false))
+                        }
+                     })
+                     .catch((error) => {
+                         return {
+                            statusCode: error.status,
+                            statusText: error
+                        }
+                     });
+                }
+                else if(parseInt(data, 10) == 0)
+                {
+                    const message:string = "No results were found."
+                    const limitExceededError = {
+                        limit: 0,
+                        message: message,
+                        status: 500
+                    }
+                    return {
+                        statusCode: res.status,
+                        statusText: limitExceededError
+                    }
+                } 
+                else if(parseInt(data, 10) > 1000){
```

Comment:
might as well just make this an else statement

Reply:
Changed to else statement.